ding_periodical.module

```diff
@@ -38,17 +38,17 @@ function ding_periodical_menu() {
  *
  * @param TingEntity $entity
  *   Ting entity for the periodical.
- * @param string $local_id
- *   Local library id (faust) for the issue.
- * @param string $volume
- *   Volume number (for year books the year).
- * @param string $issue
- *   The issue in the volume to reserve.
+ * @param string $encoded_id
+ *   Encoded string made of local id, volume and issue data of the item.
+ *   Value base64 hash of each value, glued together using '$' delimiter
+ *   and encoded again as whole.
  *
  * @return array
  *   Render array with Ajax commands.
  */
-function ding_periodical_reserve_ajax($entity, $local_id, $volume, $issue) {
+function ding_periodical_reserve_ajax($entity, $encoded_id) {
+  $id = base64_decode($encoded_id);
+  list($local_id, $volume, $issue) = array_map('base64_decode', explode('$', $id));
```

Comment:
With base64_encoded arguments we do not have to reduce to number of parameters to one here and explode it.  The only thing needed is decoding the three parameters.